runtime/compiler/trj9/control/J9Options.cpp

```diff
@@ -2234,9 +2234,13 @@ bool J9::Options::feLatePostProcess(void * base, TR::OptionSet * optionSet)
    // we remove signals. Set the noResumableTrapHandler option to note this
    // request.
    // Also disable the packed decimal part of DAA because some PD instructions
-   // trigger hardward exceptions. A new option has been added to disable traps
+   // trigger hardware exceptions. A new option has been added to disable traps
    // Which allows the disabling of DAA and traps to be decoupled from the handler
-   if (javaVM->sigFlags & J9_SIG_XRS)
+   // Multiple variants of -Xrs option are available now:
+   // -Xrs Ignore all signals (J9_SIG_XRS_SYNC, J9_SIG_XRS_ASYNC)
+   // -Xrs:sync Ignore synchronous signals (J9_SIG_XRS_SYNC)
+   // -Xrs:async Ignore asynchronous signals (J9_SIG_XRS_ASYNC)
+   if (J9_ARE_ALL_BITS_SET(javaVM->sigFlags, J9_SIG_XRS_SYNC | J9_SIG_XRS_ASYNC))
```

Comment:
The code should stay the same as it was and just check J9_SIG_XRS_SYNC. The noResumableTrapHandler uses a synchronous signal.

Reply:
fixed.